nacc/uds3/filters.py

```diff
@@ -17,28 +23,51 @@
 
 fill_non_blank_values = { 'adcid' : '41' }
 
-def write_headers(reader, output):
-    if output.fieldnames is None:
-        # Initially empty file. Write column headers.
-        output.fieldnames = reader.fieldnames
-        output_header = dict((h,h) for h in reader.fieldnames)
-        output.writerow(output_header)
-
 def filter_clean_ptid(input_ptr, filter_meta, output_ptr):
+# TODO  To deal with M Flag in Current_db.csv
 
     reader = csv.DictReader(input_ptr)
     output = csv.DictWriter(output_ptr, None)
     write_headers(reader, output)
 
-    with open(filter_meta, 'r') as ptid_file:
-        ptids = ptid_file.read().splitlines()
-        for record in reader:
-            ptid = record['ptid']
-            if ptid not in ptids:
+    for record in reader:
+        ptid = record['ptid']
+        visit_num = record['visitnum']
+        with open(filter_meta, 'r') as ptid_file:
+
+            curr_ptid = csv.DictReader(ptid_file)
+            repeat_flag = 0
+
+            for row in curr_ptid:
+                packet_type = row['Packet type']
+                curr_visit = row['Visit Num']
+
+                if ptid == row['Patient ID']:
+                    prog_followup_visit = re.compile("followup.*")
+                    prog_initial_visit = re.compile("initial.*")
+                    prog_mile_visit = re.compile("milestone.*")
+                    if packet_type == "I" and prog_initial_visit.match(record['redcap_event_name']):
+                        repeat_flag = 1
+                        print >> sys.stderr, 'Eliminated ptid : ' + ptid + " Event Name : " + record['redcap_event_name']
+
+                    elif packet_type == "F" and prog_followup_visit.match(record['redcap_event_name']):
+                        if (not visit_num and visit_num == curr_visit) or visit_num == '':
+                            repeat_flag = 1
+                            print >> sys.stderr, 'Eliminated ptid : ' + ptid + " Event Name : " + record['redcap_event_name']
+
+                    elif packet_type == "M" and prog_mile_visit.match(record['redcap_event_name']):
```

Comment:
There can be more than one milestone per subject. If that's the case, how will this handle it?